### ISSUE TITLE
Extended SMILES saved from Ketcher might be invalid for RDKit

### ISSUE THREAD

**Issue by xuzuodong** [epam/ketcher#1865]
**Steps to Reproduce**
1. Click "Open..." button
2. Click "PASTE FROM CLIPBOARD"
3. Input SMILES `c1cccc(-c2ccc(Nc3cccc4c(=O)[nH]ccc34)nc2)c1`
4. Save as `Extended SMILES`, and the result would be `C1C=C(C2C=NC(Nc3c4c(c(ncc4)=O)ccc3)=CC=2)C=CC=1`
5. Go to [RDKit.js official website](https://www.rdkitjs.com/), in their online code demo, input and run: 
``` js
var smiles = "C1C=C(C2C=NC(Nc3c4c(c(ncc4)=O)ccc3)=CC=2)C=CC=1"; // generated by Ketcher in step 4
var mol = RDKitModule.get_mol(smiles);
console.log(mol.is_valid())
```

<img width="1347" alt="image" src="https://user-images.githubusercontent.com/51878637/203482074-b1225708-1b14-47bc-91ce-31db2de05332.png">

**Actual behavior**  
1. Value of `mol.is_valid()` in RDKit.js website is false.
2. Also, using RDKit to draw molecule image would fail.

**Expected behavior**
Smiles generated by Ketcher should all be valid to RDKit? 

**Ketcher version** . 
2.6.2

**Comment by AlexanderSavelyev** [comment]
yes, aromatic bonds were not converted correctly for double bond Oxygen and aromaticity is kept as atom - which should not be a case (it should be converted to ":" bonds). It is suggested to un-aromatize such structures

**Comment by paulsmirnov** [comment]
@AlexanderSavelyev - I have a report from users that led me to this issue. Could you confirm that it is the same?

Load `N#Cc1cn[nH]c1N`, it will be saved as `N#Cc1c(N)nnc1`. This SMILES string is recognized by Biovia Draw and ChemDraw, but not RDKit 2020.09.5 (Python):

    >>> m = Chem.MolFromSmiles('N#Cc1c(N)nnc1')
    [19:51:56] Can't kekulize mol.  Unkekulized atoms: 2 3 5 6 7

**Comment by paulsmirnov** [comment]
A colleague helped me with the reasoning in terms of chemistry :)

> It seems related to tautomers and aromaticity. The exported SMILES is ambiguous, and RDKit does not make extra assumptions in order to generate a valid structure, while it seems that Biovia Draw and ChemDraw do.
In our example, either nitrogen in the ring could have an H attached. N#Cc1cn[nH]c1N specifies its location, while N#Cc1c(N)nnc1 does not. When Ketcher processes the aromaticity of the ring, the location of the H is lost. RDKit does not restore the H, leading to an invalid structure. Similarly, in this GitHub issue, the original SMILES specifies a tautomeric [nH], but the information is lost when Ketcher processes the aromaticity of the molecule.

**Comment by AlexanderSavelyev** [comment]
Need to switch to indigo for smiles generation

**Comment by even1024** [comment]
The bug appears because interchange KET-format doesn't support explicit implicit hydrogens count which can be specified in bracketed SMILES atoms as a virtual hydrogens counter. Typically it's not an issue but there are special cases when the standard valence model fails to determine the number of suppressed hydrogens. For instance In the example above, N-atom is connected to aromatic ring, so the automatic hydrogen counting is not possible. To avoid the ambiguousness [nH] explicitly specifies the number of implicit hydrogens = 1 for the nitrogen atom. To fix the issue on the ketcher's side:

1) add implicitHCount field to the atom entity of ket-format schema:

        "hImplicitCount": {
          "type": "integer",
          "enum": [0, 1, 2, 3, 4, 5]
        },

2) As ketcher has own parser/generator of MOL V2000, corresponding conversion of virtual hydrogens counter hImplicitCount to the "chemaxon style" Data S-Group should be implemented. I.e. if a MOL V2000 file has a data group as below:

M  STY  1   1 DAT
M  SLB  1   1   1
M  SAL   1  1  18
M  SDT   1 MRV_IMPLICIT_H                                                       
M  SDD   1     0.0000    0.0000    DA    ALL  1       1  
M  SED   1 IMPL_H1

it should be converted to an atom's property implicitHCount and for generating of MOL V2000 the data S-Groups should be added basing on the implicitHCount value. 

Some info about MRV_IMPLICIT_H data s-group:

http://www.scfbio-iitd.res.in/software/utility/marvin_new/marvin/help/FF/Chemaxon-specific-information-in-MDL-MOL-files_19693843.html

**Comment by KonstantinEpam23** [comment]
Functionality for supporting implicit hydrogens for mol v2000 format will be implemented separately as part of #2500
